src/material-experimental/mdc-form-field/_form-field-theme.scss

```diff
@@ -75,7 +75,10 @@
     // MDC uses the `subtitle1` level for the input label and value, but the spec shows `body1` as
     // the correct level.
     .mat-mdc-form-field-control,
-    .mat-mdc-form-field label,
+    // We have to target the inner <mat-label> rather than the <label> because MDC applies
+    // high-specificity rules that incorrectly hard-code the size as 1rem on the label.
+    .mat-mdc-form-field mat-label,
+    .mat-mdc-form-field mat-label::after,
```

Comment:
Should this be `.mdc-floating-label::after`. The asterisk is a pseudo-element of the parent element, not `mat-label`